js/stdlib/ts/ALGO.ts

```diff
@@ -1426,13 +1433,10 @@ export const connectAccount = async (networkAccount: NetworkAccount): Promise<Ac
 
         // Read map data
         const getLocalState = async (a:Address): Promise<AppStateKVs|undefined> => {
-          const ai = await getAccountInfo(a);
-          debug(`getLocalState`, ai);
-          const alss = ai['apps-local-state'] || [];
-          const fmtApplicationID = bigNumberToBigInt(ApplicationID);
-          const als = alss.find((x) => (x.id === fmtApplicationID));
-          debug(`getLocalState`, als);
-          return als ? als['key-value'] : undefined;
+          const client = await getAlgodClient();
+          const query = client.accountApplicationInformation(a, bigNumberToNumber(ApplicationID)) as unknown as ApiCall<AccountApplicationInfo>;
```

Comment:
You need to have a fallback that doesn't use `algod` and can work with just the indexer. You will test by setting the NODE_WRITE_ONLY to true and having a new function that is like `getAccountInfo` and will dispatch to either one.